File: frontend/src/types/history.ts
import type { VideoMetadata } from './video'

export interface SavedVideoItem {
    video: VideoMetadata
    saved_at: string
}

export interface QueryHistoryItem {
    video_id: string
    query: string
    relevance_score: number | null
    answer: string | null
    created_at: string
}

export interface HistoryResponse {
    saved_videos: SavedVideoItem[]
    recent_queries: QueryHistoryItem[]
}

File: backend/app/schemas/history.py
from typing import List, Optional
from pydantic import BaseModel

from app.schemas.video import VideoMetadata


class SavedVideoItem(BaseModel):
    video: VideoMetadata
    saved_at: str


class QueryHistoryItem(BaseModel):
    video_id: str
    query: str
    relevance_score: Optional[float]
    answer: Optional[str]
    created_at: str


class HistoryResponse(BaseModel):
    saved_videos: List[SavedVideoItem]
    recent_queries: List[QueryHistoryItem]